newton/_src/viewer/viewer_usd.py

```
@@ -259,14 +271,80 @@ def log_mesh(
+
+        if isinstance(texture, np.ndarray):
+            texture_dir = os.path.join(os.path.dirname(self.output_path), "textures")
+            os.makedirs(texture_dir, exist_ok=True)
+            safe_name = name.replace("/", "_").replace("\\", "_").strip("_")
+            texture_file = os.path.join(texture_dir, f"{safe_name}.png")
+            try:
+                from PIL import Image
+
+                Image.fromarray(texture).save(texture_file)
+                return os.path.abspath(texture_file)
+            except Exception:
```

Ruff (0.15.4)

[warning] Do not catch blind exception: `Exception`

(BLE001)